### DISCUSSION TITLE
Menus query

### DISCUSSION THREAD

**Discussion by Kashuab** [Shopify/storefront-api-feedback#194] [Ideas & Requests]
Since Shopify's opened up access for non-sales-channel apps, we're looking to utilize the Storefront API for our [mega menu app](https://apps.shopify.com/meteor-mega-menus?search_id=501285f0-00cf-4b00-8abf-760958dd3382&surface_detail=Meteor&surface_inter_position=1&surface_intra_position=4&surface_type=search).

We want to offer the ability for a merchant to import their Shopify navigation menus so we can display them on the storefront. However, one of the problems we're facing is the lack of a `menus` query to get access to all of their menus. Sure, we could do `query(handle: "main-menu")` however I'm sure there are merchants out there that may have created a new menu on their own.

Currently we have to rely on an app proxy which isn't quite ideal (password protected, rate limits/bot protection from our server hitting many storefronts.) It would be great if we could query for a list of their menus for better performance and reliability.

Thanks for your consideration!

**Comment by johncraigcole** [comment]
Hey @Kashuab 👋 .  I added an issue on our end to track this enhancement, but I can't share a timeline on when/if this will be added to the Storefront API.

**Reply by Kashuab** [reply]
Awesome, thank you! Totally understandable.

**Comment by thosakwe** [comment]
Are there any plans to add `createMenu`/`updateMenu` mutations?
